walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,83 @@
 // limitations under the License.
 
 package com.google.sps;
+import java.util.*;
 
-import java.util.Collection;
-
+/*
+ * Returns available meeting times upon input of a collection of pre-existing meetings and a 
+ * potential meeting time.
+ */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    // Objects containing available and booked times for mandatory and optional attendees
+    List<TimeRange> mandatoryAvailable = new ArrayList<TimeRange>();
+    List<TimeRange> mandatoryBooked = new ArrayList<TimeRange>();
+    List<TimeRange> optionalAvailable = new ArrayList<TimeRange>();
+    List<TimeRange> optionalBooked = new ArrayList<TimeRange>();
+
+    // Objects containing mandatory and optional attendees
+    Collection<String> meetingAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    
+    // Potential meeting length
+    long meetingDuration = request.getDuration();
+
+    // Returns an empty list if meeting lasts longer than 24 hours
+    if (meetingDuration > TimeRange.END_OF_DAY) {
+      return mandatoryAvailable;
+    }
+
+    // Iterates through all events and checks if attendees are already booked
+    for (Event event : events) {
+      if (!Collections.disjoint(meetingAttendees, event.getAttendees())) {
+        mandatoryBooked.add(event.getWhen());
+        optionalBooked.add(event.getWhen());
+      } else if (!Collections.disjoint(optionalAttendees, event.getAttendees())) {
+        optionalBooked.add(event.getWhen());
+      }
+    }
+
+    // Uses helper method to find available times for optional and mandatory attendees
```

Comment:
Nice job with the use of comments & helper functions! Makes your code easy to read :)